src/main/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/services/GoalService.kt

```diff
@@ -49,15 +49,38 @@ class GoalService(
       areaOfNeed = areaOfNeedEntity,
       targetDate = goal.targetDate,
       plan = planEntity,
-      relatedAreasOfNeed = relatedAreasOfNeedEntity,
+      relatedAreasOfNeed = relatedAreasOfNeedEntity.toMutableList(),
       goalOrder = highestGoalOrder + 1,
-
     )
     val savedGoalEntity = goalRepository.save(goalEntity)
 
     return savedGoalEntity
   }
 
+  @Transactional
+  fun updateGoalByUuid(goalUuid: UUID, goal: Goal): GoalEntity {
+    val goalEntity = goalRepository.findByUuid(goalUuid)
+      ?: throw Exception("This Goal is not found: $goalUuid")
```

Comment:
With what intent?

Reply:
So  type of the exception conveys (more) information about what went wrong?

Reply:
After some discussion @alexw-moj and I decided that we should change the design of how all these entity-related calls should work, but we won't do that in this PR.

We will make Repository functions return non-nullable objects so that internally-thrown Exception subclasses are propagated upwards and add some additional ExceptionHandlers in `HmppsSentencePlanExceptionHandler`. This means that all of the Controller functions should have consistent behaviour rather than the current hodgepodge.

I will create a new branch with the working through of that for GoalEntity for discussion.